Set JVM memory options for server applications

havocp opened the pull request.

havocp committed:
Scalariform reformat of JUnitXmlTestsListener.scala

havocp committed:
Set Java memory options for server applications

The defaults here are copied from Activator which
in turn are probably copied from somewhere else.
They are of course basically arbitrary, but the
JVM's defaults if we don't set memory options
are insufficient to run most apps.

If the user sets any memory options in the configuration,
we completely leave them alone and don't set any of
our own. So it's always possible to override.

typesafe-tools commented:
Can one of the admins verify this patch?

henrikengstrom commented:
Nice one!

jsuereth commented on launch/src/main/scala/xsbt/boot/ServerApplication.scala:
```diff
@@ -211,6 +208,63 @@ object ServerLauncher {
+      val stderr = new java.io.LineNumberReader(new java.io.InputStreamReader(process.getErrorStream))
+      // Looking for the first line which is `java version "1.7.0_60"` or similar
+      val lineOption = try Option(stderr.readLine()) finally stderr.close()
+      val re = """java version "[0-9]+\.([0-9]+)\..*".*""".r
```
Sorry to do this, but did you check how much larger the launcher gets if you add in regex patterns like this?

havocp replied:
no - I'm sure I could just use the plain java regex API, is that better?

jsuereth commented:
Nice, this LGTM, caveat the one check on JAR size, and travis CI passing.

havocp committed:
Ignore "java -version" IOException when launching servers

process.start and reading from the stderr stream could in
theory throw IOException, we want to return None then
rather than crashing.

havocp committed:
Server launcher: Print errors running java -version to stderr

We forward errors from the spawned server to stderr, so we are
already using stderr and this should save someone some pain
vs. just swallowing the fail.

havocp committed:
Use java Pattern directly in launcher instead of Scala API

This keeps the launch jar smaller

havocp commented:
Fixed up to use java Pattern and also to ignore (but print) errors from java -version.

havocp commented:
Travis appears to be running out of ram. Don't know if any of those failures are "real"

eed3si9n commented:
Restarted travis job.
